### ISSUE TITLE
hyperparameter definition

### ISSUE THREAD

**Issue by fmigneault** [crim-ca/dlm-extension#14]
When the model being described needs to provide hyperparameters (for example, the number of estimators used in a scikit-learn model, or a learning rate for an optimizer), there is no clear way currently to define them.

One proposal would be to have a definition as the following under a STAC Item:
```json
{
  "properties": {
    "mlm:hyperparameters": [
      {
        "name": "n_estimators",
        "value": 2
      }
    ]
  }
}
```

The `mlm:hyperparameters` could also be allowed under a specific Asset, in case the user wants to be more specific about which Asset they apply to.
```json
{
  "assets": {
    "source": {
      "title": "Model training pipeline",
      "href": "<URI-to-python-training-script>",
      "type": "text/x-python",
      "mlm:hyperparameters": [
        {
          "name": "n_estimators",
          "value": 2
        }
      ],
      "roles": [
        "metadata",
        "code",
        "model"
      ]
    } 
  }
}
```
